scapy/layers/ipsec.py

```diff
@@ -127,8 +130,47 @@ class ESP(Packet):
 
 bind_layers(IP, ESP, proto=socket.IPPROTO_ESP)
 bind_layers(IPv6, ESP, nh=socket.IPPROTO_ESP)
-bind_layers(UDP, ESP, dport=4500)  # NAT-Traversal encapsulation
-bind_layers(UDP, ESP, sport=4500)  # NAT-Traversal encapsulation
+
+
+class NAT_T(Packet):
+    """
+    UDP Encapsulation of IPsec ESP Packets [RFC3948] (for NAT-Traversal)
+    """
```

Comment:
Looking at a Wireshark dump I noticed two things which I would like to change about my current implementation of  [UDP Encapsulation of IPsec ESP Packets [RFC 3948]](https://www.rfc-editor.org/rfc/rfc3948):

![grafik](https://user-images.githubusercontent.com/11467854/198417091-41a2225c-3835-4636-b104-fdac9918beee.png)

**Layer Name**

Wireshark uses the name `UDPENCAP` instead of `NAT_T`, so I decided that it would be better to use the same name. Not only for consistency, but also because the former name seems more fitting, it emphasizes more how the packet is built, not what it is used for.

**Nat-Keepalive**

I noticed that I missed to implement a third case apart from IKE and ESP encapsulation, which is called NAT-Keepalive ([RFC 3948, Section 2.3](https://www.rfc-editor.org/rfc/rfc3948#section-2.3)), see screenshot, marked in orange. I would like to add it so the PR is complete.

![grafik](https://user-images.githubusercontent.com/11467854/198416802-792143e6-782a-4ee3-8a23-72f0ff9304ba.png)